Frameworks/VMFleet/VMFleet.psm1

```diff
@@ -8665,4 +9136,152 @@ function SetCacheBehavior
+        [switch]
+        $ResetSalt,
+
+        [Parameter()]
+        [string]
+        $Cluster = "."
+    )
+
+    if($PSBoundParameters.Count -eq 0){
+        Write-Host "Set atleast one property."
+        return;
+    }
+    if($PSBoundParameters.ContainsKey("ResetSalt")) {
+        $PSBoundParameters.Remove('ResetSalt') | Out-Null
+    }
+    $controlPath = Get-FleetPath -PathType Control $Cluster
+    $f = Join-Path $controlPath "arc.json"
+    $arcConfig = Get-ArcConfig $Cluster
+    if(!$arcConfig) {
+        $arcConfig =  new-object PSCustomObject -Property $PSBoundParameters
+    }
+    else {
+        foreach($psbp in $PSBoundParameters.GetEnumerator())
+        {
+            $arcConfig.($psbp.Key) = $psbp.Value
+        }
+    }
+    # Set default values for non-mandatory fields
+    if($arcConfig.psobject.properties.match('StoragePathName').Count -eq 0) {
+        $arcConfig | Add-Member -MemberType NoteProperty -Name 'StoragePathName' -Value $StoragePathName
+    }
+    if($arcConfig.psobject.properties.match('ImageName').Count -eq 0)  {
+        $arcConfig | Add-Member -MemberType NoteProperty -Name 'ImageName' -Value $ImageName
+    }
+    if($arcConfig.psobject.properties.match('Enabled').Count -eq 0)  {
+         $arcConfig | Add-Member -MemberType NoteProperty -Name 'Enabled' -Value $Enabled  # disabled by default
+    }
+    if(!(Test-ArcConfig $arcConfig)){
+        return;
+    }
+    $generateSalt = ($arcConfig.psobject.properties.match('Salt').Count -eq 0) -or $ResetSalt
+    if($generateSalt -or $PSBoundParameters.ContainsKey("ResourceGroup")) {
+        $retry = 0;
+        while ($retry -lt 3) {
+            if($generateSalt) {
+                $salt = Get-Salt
+            }
+            else {
+                $salt = $arcConfig.Salt
+            }
+            $nodes = @(Get-ClusterNode)
+            $isValidSalt = Invoke-CommonCommand $nodes[0] -InitBlock $CommonFunc -ScriptBlock {
+               InitializeAndGetArcHCIExtendedLoc $using:arcConfig.AzureRegistrationUser $using:arcConfig.AzureRegistrationPassword $using:arcConfig.ResourceGroup
+               $vmList = az stack-hci-vm list --resource-group $using:arcConfig.ResourceGroup | ConvertFrom-Json
+               $reg = "^vm-.{1,}-" + $using:salt + "-\d{3}$"
+               $vms = $vmList | Where-Object { $_.name -match $reg}
+               if(!$vms) { return $true }
+               return false;
+            }
+            if(!$isValidSalt) {
+                $retry++
+            }
+            else {               
+                $arcConfig | Add-Member -MemberType NoteProperty -Name 'Salt' -Value $salt -Force
+                break
+            }
+        }
+        if($retry -eq 4) {
+            throw "Unable to create a unique salt with 3 retries. Pick another resource group or run Set-ArcConfig again."
+        }
+    }
+
+    $arcConfig | ConvertTo-Json | Set-Content -Path $f -Force:$true
+
+    Write-Host "Configuration updated in $f"
```

Comment:
Is this needed? Normal assumption would be that no error/throw is success.

Reply:
Intention behind adding this was to make user aware of the path where this file is stored.

Reply:
Hi, just my 2 cents (as I tested it last night), knowing the path to the config file is good to know.